geopandas/geodataframe.py

```diff
@@ -1460,8 +1460,20 @@ def __finalize__(self, other, method=None, **kwargs):
                 object.__setattr__(self, name, getattr(other.left, name, None))
         # concat operation: using metadata of the first object
         elif method == "concat":
+            reference_obj = other.objs[0]
+            geosrs_metadata_mapping = {"_crs": "crs", "_geometry_column_name": "name"}
+
             for name in self._metadata:
-                object.__setattr__(self, name, getattr(other.objs[0], name, None))
+                if isinstance(reference_obj, GeoSeries):
+                    # propagate geo col info post series concat
+                    # (column names are dummy values inside _constructor_expanddim)
+                    name_alias = geosrs_metadata_mapping.get(name, name)
+                    metadata_value = getattr(reference_obj, name_alias, None)
+                    if name == "_geometry_column_name" and metadata_value is None:
+                        metadata_value = 0
+                else:
+                    metadata_value = getattr(reference_obj, name, None)
+                object.__setattr__(self, name, metadata_value)
```

Comment:
This set of changes is not strictly speaking necessary for adding `_constructor_expanddim` / fixing `to_frame`, right? 
As I understand, this is only for the case of concatting GeoSerieses? 

In order to get this PR merged (I think everything else is reviewed / updated / ready to merge), let's maybe leave this out, and open an issue (or PR) about it referencing this to discuss what behaviour we would like here.